### ISSUE TITLE
config_param regression in subclasses

### ISSUE THREAD

**Issue by shyouhei** [fluent/fluentd#582]
Current master breaks [riywo/fluent-plugin-hash-forward](https://github.com/riywo/fluent-plugin-hash-forward) behaviour.

### How to reproduce

```bash
bash-3.2$ git clone git@github.com:shyouhei/fluent-config-param.git
bash-3.2$ cd fluent-config-param
bash-3.2$ bundle install -j 32 --path=vendor/bundle
bash-3.2$ bundle exec fluentd -c fluentd.conf 
2015-04-21 18:39:15 +0900 [info]: reading config file path="fluentd.conf"
2015-04-21 18:39:15 +0900 [info]: starting fluentd-0.12.7
2015-04-21 18:39:15 +0900 [info]: gem 'fluentd' version '0.12.7'
2015-04-21 18:39:15 +0900 [info]: gem 'fluent-plugin-hash-forward' version '0.3.5'
2015-04-21 18:39:15 +0900 [info]: using configuration file: <ROOT>
  <match **>
    type copy
    <store>
      type hash_forward
      heartbeat_type none
    </store>
  </match>
</ROOT>
2015-04-21 18:39:15 +0900 [info]: adding match pattern="**" type="copy"
2015-04-21 18:39:15 +0900 [error]: config error file="fluentd.conf" error="forward output heartbeat type should be 'tcp' or 'udp'"
2015-04-21 18:39:15 +0900 [info]: process finished code=256
2015-04-21 18:39:15 +0900 [warn]: process died within 1 second. exit.
bash-3.2$ 
```

Here, the error was from [lib/fluent/plugin/out_forward.rb:47](https://github.com/fluent/fluentd/blob/bdf1f4f104c00a791aa94dc20087fe2011e1fd83/lib/fluent/plugin/out_forward.rb#L47) but the `type` is not forward but `hash_forward`.  It _does_ support `none` heartbeat type [here](https://github.com/riywo/fluent-plugin-hash-forward/blob/ddbbca55848428af405a41e984d549af7c1a8536/lib/fluent/plugin/out_hash_forward.rb#L21).

So the `config_param` in the subclass is now magically ignored.

**Comment by tagomoris** [comment]
fixed at #583

**Comment by shyouhei** [comment]
:beers: awesome
